packages/@aws-cdk/aws-lambda-nodejs/lib/util.ts

```diff
@@ -88,3 +88,28 @@ export function exec(cmd: string, args: string[], options?: SpawnSyncOptions) {
 
   return proc;
 }
+
+/**
+ * Extract dependencies from a package.json
+ */
+export function extractDependencies(pkgPath: any, modules: string[]): { [key: string]: string } {
+  const dependencies: { [key: string]: string } = {};
+
+  // Use require for cache
+  const pkgJson = require(pkgPath); // eslint-disable-line @typescript-eslint/no-require-imports
+
+  const pkgDependencies = {
+    ...pkgJson.dependencies ?? {},
+    ...pkgJson.devDependencies ?? {},
+    ...pkgJson.peerDependencies ?? {},
+  };
+
+  for (const mod of modules) {
+    if (!pkgDependencies[mod]) {
+      throw new Error(`Cannot extract version for ${mod} in package.json`);
```

Comment:
Will look into this. Maybe the right way to do this is to use `npm list` or `yarn list` and get the versions from there.

Reply:
Can you detail your use case with `aws-xray-sdk-core`? Why do you need to exclude `pkginfo` in this case?

Reply:
when I try to bundle `aws-xray-sdk-core` esbuild warns:
```
node_modules/pkginfo/lib/pkginfo.js:104:15: warning: This call to "require" will not be bundled because the argument is not a string literal
    contents = require(dir + '/package.json');
```
so it will not be able to also bundle content from `require(dir + '/package.json');` and it could lead to unpredicted behavior, so I better exclude it and put the package `pkginfo` with its dependencies as it is in `node_modules` to avoid unpredicted behavior

Reply:
due to this `require` condition, I also have to exclude 'import-from' for example:
```
node_modules/import-from/index.js:4:46: warning: This call to "require" will not be bundled because the argument is not a string literal
...ts = (fromDirectory, moduleId) => require(resolveFrom(fromDirectory, modul...
                                     ~~~~~~~
node_modules/import-from/index.js:8:9: warning: This call to "require" will not be bundled because the argument is not a string literal
    return require(resolveFrom(fromDirectory, moduleId));
```

Reply:
and also in some cases `esbuild` cannot transpile some `esnext` features due to limitation, for example:
```
node_modules/@graphql-tools/load/index.cjs.js:44:8: error: Transforming for-await loops to the configured target environment is not supported yet
    for await (const loader of options.loaders) {
```
so I also excluding it

Reply:
these are all inherited dependencies, not my project directly

Reply:
OK, got it. But why don't you set `aws-xray-sdk-core` as external then? Is there a big difference?

Reply:
yes, it would work, `aws-xray-sdk-core` is not so heavy package. But we should also take into account that it will bring also these packages along: 
```
        "continuation-local-storage": "^3.2.0",
        "moment": "^2.15.2",
        "pkginfo": "^0.4.0",
        "semver": "^5.3.0",
        "underscore": "^1.8.3",
        "winston": "^2.2.0"
```
so it could lead to a tangible difference

Reply:
OK, I think the best solution is to parse the result of `npm list <mod1> <mod2> --depth 0 --json` or `yarn list --pattern "<mod1>|<mod2>" --depth=0 --json`.

The JSON format is different for npm and yarn.

I suggest doing this in another PR.

Reply:
Agreed